### DISCUSSION TITLE
quickadd capture action for inserting / replacing events

### DISCUSSION THREAD

**Discussion by pcause** [open-horizon-labs/obsidian-ics#86] [Show and tell]
I update my calendar during the say and want to update the events in my day note that I've previously featched when the note was created using templater.  I created this [quickadd ](https://quickadd.obsidian.guide/docs/) capture action for doing that.  Select the existing events block and run this action. If noting is selected should insert at the cursor.   Downside is that events previously marked as completed are now back to initial state.

create a capture action, only enable the "capture to current file" and the "capture format".  In the "capture format" box, add this:

```js quickadd
var events = await app.plugins.getPlugin('ics').getEvents(moment().format('YYYY-MM-DD'));
var tR;="\n":
events.sort((a,b) => a.utime - b.utime).forEach((e) => {
    const calendar = e.icsName.replace(" ","-");
    const callUrl = e.callUrl? ` [${e.callType}](${e.callUrl})` : '';
    const location = e.location? ` 📍 *(${e.location})*` : '';
    tR+=`- [ ] [startTime:: ${e.time}] [endTime:: ${e.endTime}] **${e.summary}** ${location}${callUrl} \n`
    if (e.description && e.description != "" && e.description != "\n") {
        const inlineDescription = e.description.replace(/(?:\r\n|\r|\n)/g," ").replace(/\s{2,}/g," ");
        tR+=`> [!INFO]-\n>    - ${inlineDescription}\n`
    }
})
return tR;
```

NOTE:  I put the description into a callout.  I can have a lot of things in the description, especially Zoom meeting invitation, and this helps make the events more readable.  you can  edit the format line to have things meet you needs.

**Comment by nestor50** [comment]
Hi @pcause, that's exactly what I had been looking for but, unfortunately, what I get when I run the quick add capture is just the code pasted into the note; do you have any suggestion? Thanks a lot 
Nestorito

**Reply by pcause** [reply]
did you have the quickadd js indicators in the capture box?  so you start with:

>  \`\`\`  js quickadd

then have the code above and then end with


> \`\`\`

**Reply by nestor50** [reply]
Thanks @pcause but no luck! 
What I did is:
- create a capture choice
- set it as follows:
> - capture to active file
> - pasted the code inside the js QuickAdd 'clause' (```js quickadd .... ```) and the code
`

var events = await app.plugins.getPlugin('ics').getEvents(moment().format('YYYY-MM-DD'));
var tR;="\n":
events.sort((a,b) => a.utime - b.utime).forEach((e) => {
    const calendar = e.icsName.replace(" ","-");
    const callUrl = e.callUrl? ` [${e.callType}](${e.callUrl})` : '';
    const location = e.location? ` 📍 *(${e.location})*` : '';
    tR+=`- [ ] [startTime:: ${e.time}] [endTime:: ${e.endTime}] **${e.summary}** ${location}${callUrl} \n`
    if (e.description && e.description != "" && e.description != "\n") {
        const inlineDescription = e.description.replace(/(?:\r\n|\r|\n)/g," ").replace(/\s{2,}/g," ");
        tR+=`> [!INFO]-\n>    - ${inlineDescription}\n`
    }
})
return tR;
```
`
In my note:
- I select the text of the events 
- invoke from command menu the "ics refresh" quickadd action
but it results in the code being pasted in the note. Should I activate some option in quickadd or something alike?
Thanks!

**Reply by muness** [reply]
@nestor50 - do you the Templater plugin installed and enabled?

**Reply by nestor50** [reply]
Yes @muness, Templater is installed and enabled and I haven't noticed any problems. 
if the js indicator is ````js quickadd```` with no space between the dashes and "js" what I get is a syntax error: "Unexpected token '=' "; that is what I see in the console. I also noticed that I have a "ICS: styles.css not found" but I don't know if it is related to the same issue
thanks for your time

**Reply by nestor50** [reply]
Maybe another approach to have events autoupdated would be using the dataviewjs code provided in the readme file:
```
```dataviewjs
var events = await app.plugins.getPlugin('ics').getEvents("{{date:YYYY-MM-DD}}");
var mdArray = [];
events.forEach((e) => {
  mdArray.push(`${e.time} ${e.summary} ${e.location}: ${e.description}`.trim())
})
dv.list(dv.array(mdArray))
```


 Is it possible to have events formatted with the checklist at the beginning ( "- [ ]") and with the name of the event in bold?
